node/network/statement-distribution/src/lib.rs

```diff
@@ -115,16 +114,19 @@ const LOG_TARGET: &str = "parachain::statement-distribution";
 const MAX_LARGE_STATEMENTS_PER_SENDER: usize = 20;
 
 /// The statement distribution subsystem.
-pub struct StatementDistributionSubsystem {
+pub struct StatementDistributionSubsystem<R: rand::Rng> {
```

Comment:
```suggestion
pub struct StatementDistributionSubsystem<R> {
```

Reply:
Usually bounds are avoided in struct generics, unless they can't be expressed as part of the body.